clink/app/src/host/host.cpp

```diff
@@ -496,9 +496,12 @@ bool host::edit_line(const char* prompt, str_base& out)
         {
             // If the line is a directory, rewrite the line to invoke the CD
             // command to change to the directory.
-            if (!m_doskey.resolve(out.c_str(), m_doskey_alias))
+            m_doskey.resolve(out.c_str(), m_doskey_alias);
```

Comment:
I believe I understand the intent:  to allow a doskey macro `..` or `-` to do something other than these new default behaviors.

I think it's worth restructuring the code here. I inserted lines 495 through 505 here, but on further examination I think maybe they belong outside the `while (1)` loop. And it seems like `intercept_directory` should be called after resolving any doskey aliases, so that for example one could create an alias like `doskey prevdir=-`.

Maybe something more like
```cpp
490:
            // Add the line to the history.
            if (m_history)
                m_history->add(out.c_str());
        }
        break;
    }

    if (!resolved)
    {
        m_doskey.resolve(out.c_str(), m_doskey_alias);
        if (m_doskey_alias)
            out.clear();
    }

    if (ret)
    {
        // If the line is a directory, rewrite the line to invoke the CD
        // command to change to the directory.
        intercept_directory(out);
    }

    s_history_db = nullptr;
```

Reply:
I've made the above change and I think it achieves what was desired, plus some additional benefits.

Reply:
a0da1a5e0d51f87f3d836910dbf0e3bcc599ae82 addresses this motivation for the PR, in a little different manner.